### DISCUSSION TITLE
How to use terraform functions with terranix (like `base64decode`)

### DISCUSSION THREAD

**Discussion by das-monki** [terranix/terranix#95] [Q&A]
I've tried to move my terraform config to terranix, also to force myself to get more familiar with nix.

Now I'm stuck translating the following block, thankful for any help!

```tf
provider "google-beta" {
  project     = local.project
  region      = local.region
  zone        = local.zone
  credentials = base64decode(google_service_account_key.firebase_key.private_key)
}
```

I've tried with `credentials = "\${base64decode(google_service_account_key.firebase_key.private_key)}";` but got following error:
```
Error: unable to parse credentials
│
│   with provider["registry.terraform.io/hashicorp/google-beta"],
│   on config.tf.json line 20, in provider.google-beta:
│   20:     }
│ 
│ unexpected end of JSON input
```

**Comment by mrVanDalo** [comment]
Hmm, that is a tough one. I can only guess that the content is not properly decoded.
https://github.com/hashicorp/hcl/blob/main/json/spec.md#static-call < maybe this helps?
